chore(OSC-1274): Update best practices for batching (#6683)
SumoLogic/sumologic-documentation: docs/send-data/kubernetes/best-practices.md

[filling_up_alert]: https://runbooks.prometheus-operator.dev/runbooks/kubernetes/kubepersistentvolumefillingup/
[sumo-otelcol-batching-doc]: ../../opentelemetry-collector/data-source-configurations/additional-configurations-reference/#using-batch-processor-to-batch-data

### Batching Migration
We are migrating the default batching logic from the batch processor to exporter-side batching to improve service reliability. For more details refer [Link](https://github.com/open-telemetry/opentelemetry-collector/issues/8122)

#### Queue Size

> When an exporter is preceded by a batch processor, `sending_queue.queue_size` represents the number of batches produced by the batch processor. With exporter-side batching, the batch processor is removed, so `sending_queue.queue_size` instead represents the number of individual requests. As a result, if the queue size is not adjusted accordingly, the queue may fill up more quickly and potentially lead to data loss.

Reviewer: `sending_queue.queue_size` instead represents the number of individual **requests**
Not individual requests, individual records

To avoid this, it is recommended to increase `sending_queue.queue_size` by a factor equal to the batch processor’s `send_batch_size`.

##### Example configuration
Assuming we have a batch processor with config
```yaml
batch:
  send_batch_size: 1_024
  timeout: 1s
  send_batch_max_size: 2_048
```

We have the `sending_queue.queue_size` configured as,
```yaml
metadata:
  logs:
    config:
      merge:
        exporters:
          otlphttp:
            sending_queue:
              queue_size: 1000
```

The new config should be,
```yaml
metadata:
  logs:
    config:
      merge:
        exporters:
          otlphttp:
            sending_queue:

>               queue_size: 1024000 ## = 1000 * 1024

Reviewer: ++ add steps about migrating batch settings from batch processor to exporter.

```

If you have completed the configuration changes and want to proceed with the installation, set this flag to `true` to acknowledge that you understand the changes and their impact.

```yaml

> customBatchingConfigured: true

Reviewer: Please move to sumologic.customBatchingConfigured

[filling_up_alert]: https://runbooks.prometheus-operator.dev/runbooks/kubernetes/kubepersistentvolumefillingup/
[sumo-otelcol-batching-doc]: ../../opentelemetry-collector/data-source-configurations/additional-configurations-reference/#using-batch-processor-to-batch-data

### Batching Migration

> We are migrating the default batching logic from the batch processor to exporter-side batching to improve service reliability. For more details refer [Link](https://github.com/open-telemetry/opentelemetry-collector/issues/8122)

Reviewer 2: Docs should be evergreen — this reads as a point-in-time migration announcement that will become misleading once complete.

```suggestion
### Configuring Exporter-Side Batching
Sumo Logic uses exporter-side batching instead of the batch processor. If you previously configured the batch processor, update your settings as follows. For more details, see [OpenTelemetry Collector issue #8122](https://github.com/open-telemetry/opentelemetry-collector/issues/8122).
```

— via Claude Code

#### Queue Size
When an exporter is preceded by a batch processor, `sending_queue.queue_size` represents the number of batches produced by the batch processor. With exporter-side batching, the batch processor is removed, so `sending_queue.queue_size` instead represents the number of individual records. As a result, if the queue size is not adjusted accordingly, the queue may fill up more quickly and potentially lead to data loss.

> To avoid this, it is recommended to increase `sending_queue.queue_size` by a factor equal to the batch processor’s `send_batch_size`.

Reviewer 2: ```suggestion
To avoid this, we recommend increasing `sending_queue.queue_size` by a factor equal to the batch processor’s `send_batch_size`.
```

##### Example configuration

> Assuming we have a batch processor with config

Reviewer 2: ```suggestion
If you have a batch processor configured as follows:
```

```yaml
batch:
  send_batch_size: 1_024
  timeout: 1s

>   send_batch_max_size: 2_048

Reviewer 2: should the underscores be removed for consistency with the other `send_batch_size` and `send_batch_max_size` instances?

```suggestion
  send_batch_size: 1024
  timeout: 1s
  send_batch_max_size: 2048
```

```

> We have the `sending_queue.queue_size` configured as,

Reviewer 2: ```suggestion
And the `sending_queue.queue_size` configured as:
```

```yaml
metadata:
  logs:
    config:
      merge:
        exporters:
          otlphttp:
            sending_queue:
              queue_size: 1000
```

> The new config should be,

Reviewer 2: ```suggestion
You'd need to update to the following configuration:
```

```yaml
metadata:
  logs:
    config:
      merge:
        exporters:
          otlphttp:
            sending_queue:
              queue_size: 1024000 ## = 1000 * 1024
```
#### Batch Size

> The batching logic is moved from processor to exporter so if you have custom configured the batch size like,

Reviewer 2: ```suggestion
If your batch processor has custom batch size settings like the example below:
```

```yaml
metadata:
  logs:
    config:
      merge:
        processors:
          batch:
            timeout: 5s
            send_batch_size: 5000
            send_batch_max_size: 10000
```

> The new config should be,

Reviewer 2: ```suggestion
You'd need to migrate them to the exporter configuration as follows:
```

```yaml
metadata:
  logs:
    config:
      merge:
        exporters:
          sumologic: ## for each exporter
            sending_queue:
              ## If you had the old queue value as x 
              ## the new queue_size should x * 5000
              queue_size: 5000000
              batch:
                min_size: 5000
                max_size: 10000
                flush_timeout: 5s
```

> More details of the exporter batch config can be found in [Link](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/README.md#sending-queue-batch-settings)

Reviewer 2: `[Link]` is not descriptive link text (accessibility + style). Suggestion also cuts the passive phrasing.

```suggestion
For exporter batch config details, see [Sending Queue Batch Settings](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/README.md#sending-queue-batch-settings).
```

— via Claude Code
